calc.rb

```diff
@@ -0,0 +1,83 @@
+class BasicCalculator
+	attr_accessor :num1, :num2, :op
+
+	def initialize 
+		puts "please enter your first number, operator and then second number"
+		num1 = gets.chomp.to_i
+		op = gets.chomp
+		num2 = gets.chomp.to_i
+		self.num1 = num1
+		self.num2 = num2
+		self.op = op
+	end
+	def add
+		num1 + num2
+	end
+	def subtract
+		num1 - num2
+	end
+	def multiply
+		num1 * num2
+	end
+	def divide
+		num1 / num2s
+	end
+	def output
+		case op
+		when "+" then puts "#{add}"
+		when "-" then puts "#{subtract}"
+		when "*" then puts "#{multiply}"
+		when "/" then puts "#{divide}"
+		else
+			puts "This is not a valid operation"
+		end
+	end
+end
+
+class AdvCalculator < BasicCalculator
+	def power
+		puts num1 ** num2
+	end
+	def sqrt
+		puts Math.sqrt(num1)
+	end
+	def output
+		case op
+		when "+" then puts "#{add}"
```

Comment:
This is a little blunt, I would probably user a nicer way to present this data to the user - e.g. at least say "The result is #{result}".